src/layer/const.ts

```diff
@@ -99,3 +100,54 @@ export const SH_SERVICE_HOSTNAMES_V3: string[] = [
   'https://services-uswest2.sentinel-hub.com/',
   'https://creodias.sentinel-hub.com/',
 ];
+
+export type GetStatsAndHistogramParams = {
+  fromTime: Date;
+  toTime: Date;
+  crs: CRS;
+  resolution: number;
+  geometry: Polygon;
+  binAmount?: number;
+  evalscript?: string;
```

Comment:
I'm not convinced that this is the proper way - the "layer" is a separate visualization of data. If the evalscript changes, a new layer should be instantiated.

This concept allows you to leverage tools which are already there, for example `updateLayerFromServiceIfNeeded`. Once we implement `evalscriptUrl` fetching we can use that too.

In other words, I would suggest removing `evalscript` from params and using `this.evalscript` instead.

Reply:
Sure, it is inconsistent with `getMap`, so changed to use `this.evalscript`